pkg/openstack/loadbalancer.go

```diff
@@ -605,6 +611,35 @@ func (lbaas *LbaasV2) getLoadBalancerLegacyName(_ context.Context, _ string, ser
 	return cloudprovider.DefaultLoadBalancerName(service)
 }
 
+// Returns a list of custom load balancer listener tags for the specified  service.
+func (lbaas *LbaasV2) customLoadBalancerListenerTag(service *corev1.Service, svcConf *serviceConfig) ([]string, error) {
+	var listenerTags []string
+	annotationVal := getStringFromServiceAnnotation(service, ServiceAnnotationLoadBalancerListenerTags, "")
+
+	if annotationVal == "" {
+		listenerTags = append(listenerTags, svcConf.lbName)
+		return listenerTags, nil
+	}
+
+	listOfTags := strings.Split(annotationVal, ",")
+
+	for _, tags := range listOfTags {
+		tags = strings.TrimSpace(tags)
+		tagKey, tagValue, ok := strings.Cut(tags, "=")
+		if !ok {
+			continue
+		}
+
+		if tagKey == "" || tagValue == "" {
+			return listenerTags, fmt.Errorf("invalid tagging format: %q", tags)
+		}
```

Comment:
Tags [are just arbitrary strings in Octavia](https://docs.openstack.org/api-ref/load-balancer/v2/index.html?expanded=create-listener-detail#id3). What's the point of forcing them to have `key=value` format?

Reply:
i though it will be key-value pair after reading through the opened issue 

![image](https://github.com/kubernetes/cloud-provider-openstack/assets/54990092/b2c17708-62df-43d2-8ea6-c4509fae8f42)

Reply:
It's not exactly true, it's just the custom format that might be used by the env of the person raising the issue, but tags are free-form really in Octavia.